src/grids.cpp

```diff
@@ -3356,13 +3374,20 @@ bool pj_bilinear_interpolation_three_samples(const GenericShiftGrid *grid,
     }
 
     const auto &extent = grid->extentAndRes();
-    double grid_x = (lp.lam - extent.westLon) / extent.resLon;
-    if (lp.lam < extent.westLon) {
-        grid_x = (lp.lam + 2 * M_PI - extent.westLon) / extent.resLon;
-    } else if (lp.lam > extent.eastLon) {
-        grid_x = (lp.lam - 2 * M_PI - extent.westLon) / extent.resLon;
+    if (!extent.isGeographic) {
+        pj_log(ctx, PJ_LOG_ERROR,
+               "Can only handle grids referenced in a geographic CRS");
+        pj_ctx_set_errno(ctx, PJD_ERR_FAILED_TO_LOAD_GRID);
+        return false;
+    }
+
```

Comment:
A short comment about how we're computing grid locations and what we need to do it would be a helpful echo for the future.

Reply:
done